reducer/countdistinct.go

```diff
@@ -22,8 +22,16 @@ func (cdp *CountDistinctProto) Instantiate() Interface {
 	}
 }
 
-func NewCountDistinctProto(target string, resolver expr.FieldExprResolver) *CountDistinctProto {
-	return &CountDistinctProto{target, resolver}
+func (cdp *CountDistinctProto) TargetResolver() expr.FieldExprResolver {
+	return cdp.tresolver
+}
+
+func NewCountDistinctProto(target string, tresolver, resolver expr.FieldExprResolver) *CountDistinctProto {
+	return &CountDistinctProto{
+		target:    target,
+		tresolver: tresolver,
+		resolver:  resolver,
+	}
```

Comment:
We usually put constructors immediately after the type definition, not after its methods.